client/src/components/Grid/configs/adminRoles.ts

```diff
@@ -0,0 +1,187 @@
+import { faEdit, faKey, faPlus, faTrash, faTrashRestore } from "@fortawesome/free-solid-svg-icons";
+import { useEventBus } from "@vueuse/core";
+import axios from "axios";
+
+import { deleteRole, purgeRole, undeleteRole } from "@/api/roles";
+import Filtering, { contains, equals, toBool, type ValidFilter } from "@/utils/filtering";
+import { withPrefix } from "@/utils/redirect";
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import type { ActionArray, FieldArray, GridConfig } from "./types";
+
+const { emit } = useEventBus<string>("grid-router-push");
+
+/**
+ * Local types
+ */
+type RoleEntry = Record<string, unknown>;
+
+/**
+ * Request and return data from server
+ */
+async function getData(offset: number, limit: number, search: string, sort_by: string, sort_desc: boolean) {
+    const query = {
+        limit: String(limit),
+        offset: String(offset),
+        search: search,
+        sort_by: sort_by,
+        sort_desc: String(sort_desc),
+    };
+    const queryString = new URLSearchParams(query).toString();
+    const { data } = await axios.get(withPrefix(`/admin/roles_list?${queryString}`));
+    return [data.rows, data.rows_total];
+}
```

Comment:
Not for the scope of this PR, but I wonder what would be the future approach to handle all these client routes.
- Do we still want to support these kinds of client routes or do we want to replace all of them with API ones?
- Should we create a new endpoint in the API that does what `/admin/roles_list` needs? 
- or some general "grid" endpoint to manage them all in a similar way?

Reply:
Definitely, the goal is to entirely use api routes and deprecate the legacy controller endpoints in the future. I think it would be better to use the regular resource-specific `index` endpoints like we do for visualizations and pages. Basically we would move the filter query from the controller to the corresponding manager, and then add an api endpoint. However in this first stage, my goal is to decouple and restructure all legacy grids first, such that they use Vue and API-operations, and then follow-up with moving the data request handling to api endpoints as necessary.